go/vm/evmc/evmc_steppable_interpreter.go

```diff
@@ -46,6 +46,12 @@ func (e *SteppableEvmcInterpreter) StepN(
 		context: params.Context,
 	}
 
+	evmcBlobHashes := make([]evmc.Hash, 0, state.CallData.Length())
+	for _, hash := range params.BlobHashes {
+		evmcBlobHashes = append(evmcBlobHashes, evmc.Hash(hash))
+	}
+	host_ctx.evmcBlobHashes = evmcBlobHashes
```

Comment:
I am not sure if this solves the problem, and if it does I have concerns about stability and being idiomatic:
- On the one hand host_ctx will keep the member pointer (slice) evmcBlobHashes alive over the StepN call. 
- On the other hand cgo **requires** to pin pointers passed to C, and I am afraid that they do check if the memory is pinned before the call.  

I guess it is not enough with having the variable bound and alive during C program execution because cgo wants to provide extra guarantees (what happens if the C side keeps a copy of that pointer after the C function has been called? )

Reference: https://pkg.go.dev/cmd/cgo
```
All Go pointers passed to C must point to pinned Go memory. Go pointers passed as function arguments to C 
functions have the memory they point to implicitly pinned for the duration of the call. Go memory reachable 
from these function arguments must be pinned as long as the C code has access to it. Whether Go memory is 
pinned is a dynamic property of that memory region; it has nothing to do with the type of the pointer. 

```

Reply:
Regarding readability, this is not the place to convert/copy data, neither. 
The responsibility of this function is to trigger the correct conversions and handle logic flow. Data transformations belong somewhere else.